### ISSUE TITLE
UndefinedVariable false positive with reference 

### ISSUE THREAD

**Issue by bkdotcom** [phpmd/phpmd#766]
- PHPMD version: 2.8.2
- Installation type: composer
- Operating System / Distribution & Version: OSX Mojave

### Current Behavior
Explain the result that you get.

> Avoid using undefined variables such as '$row' which will lead to PHP notices.

### Expected Behavior

No error.. the variable is used

### Steps To Reproduce:

```PHP
foreach ($data as $k => &$row) {   // error triggered here
    $row['foo'] = 'bar';                         // and here
}

```

## Checks before submitting

I apologize if this is a dupe.. don't see existing

**Comment by kylekatarnls** [comment]
Thanks, this is a duplicate of #734, it will be fixed in 2.9.0 that is about to be released. You can yet try it requiring `"phpmd/phpmd": "dev-master"`

**Comment by kylekatarnls** [comment]
If you can still have the issue with dev-master, please let me know and I will re-open.
